x-pack/plugins/infra/server/kibana.index.ts

```diff
@@ -39,7 +39,7 @@ export const initServerWithKibana = (kbnServer: KbnServer) => {
           all: ['infrastructure-ui-source'],
           read: ['config'],
         },
-        ui: ['show', 'configureSource'],
+        ui: ['show', 'configureSource', 'save'],
```

Comment:
How is "save" going to differ from "configureSource"? So far the latter was used to determine whether the user is allowed to save an updated source configuration.

Reply:
We've tried to keep the ui capabilities "purpose specific" for theoretical benefits and also to allow more flexibility for the future. It's possible that having both a `save` and a `configureSource` ui capability won't afford us any benefits going forward, it might just be future proofing for something we never fully take advantage of.

On the theoretical level, we want each of the ui capabilities to be "consumption specific". So, if we're looking to disable the ability to "configure source", we'd add this as a specific ui capability, as opposed to adding a general purpose "disable all saving related functionality". It's our hope that in the future this will allow us to add more privilege levels and optional privileges to the applications that allow us to disable certain functionality, but not others. This methodology conflicts a bit with the "read only" badge which we're now adding to the headers, which should show-up when the user isn't able to "save" anything at all. I added the separate ui capability so we weren't tying these two things together just quite yet, to hopefully allow us a tiny amount of less rework if the future plays out how anticipated.

The main thing we want to prevent is exposing a general purpose ui capability which is consumed by everything, making it hard to introduce more granularity. Given that the infra apps only allow you to save the source configuration at this point, it's largely inconsequential outside of it potentially becoming a slippery slope.

Reply:
Ok, I was just wary to use a capability as the basis for rendering which is never actually used anywhere when checking permissions. Thanks for the explanation!